src/gc.c

```diff
@@ -853,6 +854,7 @@ STATIC_INLINE void gc_setmark_big(jl_ptls_t ptls, jl_taggedvalue_t *o,
 STATIC_INLINE void gc_setmark_pool_(jl_ptls_t ptls, jl_taggedvalue_t *o,
                                     uint8_t mark_mode, jl_gc_pagemeta_t *page) JL_NOTSAFEPOINT
 {
+    assert(page);
```

Comment:
```suggestion
```
it would need to go after the MEMDEBUG else block, since we don't have pages in MEMDEBUG mode, which is causing ASAN to get upset

Worthwhile to note that it looks like I introduced the second issue in c92ce0a65589a1fc6379feeb26502a7bfe3139ea. I don't remember the thinking there, as it seems I updated `jl_gc_state_set` to be wrong, but changed `emit_gc_state_set` and `jl_eh_restore_state` to be right